.circleci/config.yml

```diff
@@ -331,20 +330,22 @@ jobs:
           command: |
             rm -rf sql/
             cp -r /tmp/workspace/generated-sql/sql sql
-      - run:
-          name: Determine docker image name
-          command:
-            # yamllint disable-line rule:line-length
-            echo 'IMAGE="${CIRCLE_PROJECT_USERNAME+$CIRCLE_PROJECT_USERNAME/}${CIRCLE_PROJECT_REPONAME:-bigquery-etl}:${CIRCLE_TAG:-latest}"' > $BASH_ENV
-      - run:
-          name: Build docker image
-          command: docker build . --pull --tag "$IMAGE"
-      - run:
-          name: Deploy to Dockerhub
-          command: |
-           echo "${DOCKER_PASS:?}" | \
-            docker login -u "${DOCKER_USER:?}" --password-stdin
-            docker push "$IMAGE"
+      - when:
+          condition: &deploy-condition
+            or:
+              - equal: [main, << pipeline.git.branch >>]
+              - << pipeline.git.tag >>
```

Comment:
If this is git tag, do we still want to publish an image with tag `latest`? Can we add some commentary here about the intention of tagging?

Reply:
the tag is defined on line 344 below: `tag: ${CIRCLE_TAG:-latest}`, so the image tag matches the git tag, and latest is used for branch main. This condition is equivalent to the one in workflows that it replaced:
```
          filters:
            branches:
              only: main
            tags:
              only: /.*/
```